ghmirror/data_structures/requests_cache.py

```diff
@@ -0,0 +1,50 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2020
+# Author: Maha Ashour <mashour@redhat.com>
+
+"""
+Implements caching backend
+"""
+
+import os
+
+from ghmirror.data_structures.monostate import InMemoryCache
+from ghmirror.data_structures.redis_data_structures import RedisCache
+
+
+class RequestsCache:
+    """
+    Instantiates either a InMemoryCache or a Redis Cache object
+    """
+    def __new__(cls, *args, store=None, **kwargs):
+        cache_type = store
```

Comment:
added an optional positional argument here so that I can instantiate the RequestCache as InMemory or Redis without setting environment variables. this should be helpful for testing.

Reply:
Why the name change? From store to cache_type?

Anyway, it's I see as a bad sign when you create "testing awareness" in your code. That will typically end up in you testing only one not-so-happy path where you don't reproduce the real execution conditions.

My suggestion here is:
- get the environment variable at module level.
- test with nothing set (`None`).
- test is with all the possible values - you can mock its value so you don't really have to set it at the system level. [Example](https://github.com/app-sre/github-mirror/blob/master/tests/functional/test_api.py#L186).

Reply:
makes sense. dropping the change.